fastlane_bot/modes/base_triangle.py

```diff
@@ -174,6 +174,141 @@ def get_combos(
                             combos,
                         )
         return combos
+    
+    def sort_pairs(self, pairs):
+        # Clean up the pairs alphabetically
+        return ["/".join(sorted(pair.split('/'))) for pair in pairs]
+
+    def flatten_nested_items_in_list(self, nested_list):
+        # unpack nested items
+        flattened_list = []
+        for items in nested_list:
+            flat_list = []
+            for item in items:
+                if isinstance(item, list):
+                    flat_list.extend(item)
+                else:
+                    flat_list.append(item)
+            flattened_list.append(flat_list)
+        return flattened_list
+    
+    def get_triangle_groups(self, flt, x_y_pairs):
+        # Get groups of triangles that conform to (flt/x , x/y, y/flt) where x!=y
+        triangle_groups = []
+        for pair in x_y_pairs:
+            x,y = pair.split('/')
+            triangle_groups += [("/".join(sorted([flt,x])), pair, "/".join(sorted([flt,y])))]
+        return triangle_groups
```

Comment:
Why not simply `return self.sort_pairs(x_y_pairs)`?

Reply:
I think you have misunderstood the function. The triangle_groups now has three pairs in the list

Reply:
I'll have another look then...